sequencing-server/src/routes/seqjson.ts

```diff
@@ -0,0 +1,461 @@
+      ).metadata;
+
+      const simulatedActivitiesForSeqId =
+        await context.simulatedActivityInstanceBySimulatedActivityIdDataLoader.loadMany(
+          activityInstanceCommandRowsForSeq.map(row => ({
+            simulationDatasetId,
+            simulatedActivityId: row.activity_instance_id,
+          })),
+        );
+
+      const simulatedActivitiesLoadErrors = simulatedActivitiesForSeqId.filter(ai => ai instanceof Error);
+
+      if (simulatedActivitiesLoadErrors.length > 0) {
+        throw new Error(
+          `Error loading simulated activities for seqId: ${seqId}, simulationDatasetId: ${simulationDatasetId}`,
+          { cause: simulatedActivitiesLoadErrors },
+        );
+      }
+
+      const sortedActivityInstances = (
+        simulatedActivitiesForSeqId as Exclude<(typeof simulatedActivitiesLoadErrors)[number], Error>[]
+      ).sort((a, b) => Temporal.Instant.compare(a.startTime, b.startTime));
+
+      const sortedSimulatedActivitiesWithCommands = sortedActivityInstances.map(ai => {
+        const row = activityInstanceCommandRows.find(row => row.activity_instance_id === ai.id);
+        // Hasn't ever been expanded
+        if (!row) {
+          return {
+            ...ai,
+            commands: null,
+            errors: null,
+          };
+        }
+        return {
+          ...ai,
+          commands: row.commands?.map(CommandStem.fromSeqJson) ?? null,
+          errors: row.errors,
+        };
+      });
+
+      const errors = sortedSimulatedActivitiesWithCommands.flatMap(ai => ai.errors ?? []);
+
+      const sequenceJson = seqBuilder(
+        sortedSimulatedActivitiesWithCommands,
+        seqId,
+        seqMetadata,
+        simulationDatasetId,
+      ).toSeqJson();
+
+      if (errors.length > 0) {
+        return {
+          status: FallibleStatus.FAILURE,
+          seqJson: sequenceJson,
+          errors,
+        };
+      } else {
+        return {
+          status: FallibleStatus.SUCCESS,
+          seqJson: sequenceJson,
+          errors: [],
+        };
+      }
+    }),
+  );
+
+  res.json(
+    promises.map(promise => {
+      if (isResolved(promise)) {
+        return promise.value;
+      } else {
+        return {
+          status: FallibleStatus.FAILURE,
+          seqJson: null,
+          errors: [promise.reason],
+        };
+      }
+    }),
+  );
+
+  return next();
+});
+
+/** Generate Sequence EDSL from sequence JSON
+ *
+ * @deprecated Use `/bulk-get-edsl-for-seqjson` instead
+ */
+seqjsonRouter.post('/get-edsl-for-seqjson', async (req, res, next) => {
+  const seqJson = req.body.input.seqJson as SeqJson;
+  const validate = new Ajv({ strict: false }).compile(schema);
+
+  if (!validate(seqJson)) {
+    throw new Error(
+      `POST /seqjson/bulk-get-edsl-for-seqjson: Uploaded sequence JSON is invalid and does not match the current spec`,
+    );
+  }
+
+  res.json(Sequence.fromSeqJson(seqJson).toEDSLString());
+
+  return next();
+});
```

Comment:
Maybe remove this one as well and remove the action:

Reply:
Let's keep these for now since they are still used by downstream tools and we don't want to introduce breaking changes with this refactor